### ISSUE TITLE
Easier settings

### ISSUE THREAD

**Issue by penrods** [ChristopherRogers1991/mycroft-hue#7]
I just tried this for the first time and really enjoyed using it.  I'd love to see it bumped up to use the latest Mycroft option setting mechanisms -- I think it will be much more user-friendly.  I love to see it using settingsmeta.json to provide a webUI for the IP address and other options, moving away from the config file approach.

I can assist if you need.

**Comment by ChristopherRogers1991** [comment]
Hey @penrods,

I actually started work on this last week (just pushed my work in progress to [feature/settingsmeta](https://github.com/ChristopherRogers1991/mycroft-hue/tree/feature/settingsmeta?files=1) if you want to take a look), but ran into a few issues before I could finish, and subsequently went out of town. I don't have a computer with me I can develop on here (responding via my phone), but plan to pick it up when I return in about a week. If you have any tips or suggestions, or want to open a PR in the meantime though, I'd definitely be open to that. Here are the problems I ran into:

https://home.mycroft.ai/#/skill just started showing:
> Sorry, you don't have skills to configure.

In the javascript console I'm getting:
```
 "The server was not able to produce a timely response to your request.\r\nPlease try again in a short while!"
```
It looks like it's trying to hit `https://api.mycroft.ai/v1/user/<UUID>/skill` when it times out.

In the Mark 1 skills log, I'm seeing:
```
03:29:29.346 - mycroft.skills.settings:_poll_skill_settings:368 - ERROR - HTTPSConnectionPool(host='api.mycroft.ai', port=443): Read timed out. (read timeout=15)
03:29:29.348 - mycroft.skills.settings:_poll_skill_settings:369 - ERROR -
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/mycroft_core-0.9.8-py2.7.egg/mycroft/skills/settings.py", line 364, in _poll_skill_settings
    self.initiatlize_remote_settings()
  File "/usr/local/lib/python2.7/site-packages/mycroft_core-0.9.8-py2.7.egg/mycroft/skills/settings.py", line 136, in initiatlize_remote_settings
    settings = self._request_my_settings(hashed_meta)
  File "/usr/local/lib/python2.7/site-packages/mycroft_core-0.9.8-py2.7.egg/mycroft/skills/settings.py", line 403, in _request_my_settings
    settings = self._request_settings()
  File "/usr/local/lib/python2.7/site-packages/mycroft_core-0.9.8-py2.7.egg/mycroft/skills/settings.py", line 419, in _request_settings
    "path": self._api_path
  File "/usr/local/lib/python2.7/site-packages/mycroft_core-0.9.8-py2.7.egg/mycroft/api/__init__.py", line 48, in request
    return self.send(params)
  File "/usr/local/lib/python2.7/site-packages/mycroft_core-0.9.8-py2.7.egg/mycroft/api/__init__.py", line 73, in send
    data=data, json=json, timeout=(3.05, 15))
  File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 499, in send
    raise ReadTimeout(e, request=request)
ReadTimeout: HTTPSConnectionPool(host='api.mycroft.ai', port=443): Read timed out. (read timeout=15)
```
This started about maybe 4 days ago (I posted in the mattermost general room a couple days ago, but didn't have an opportunity to go to the forum or create a GitHub issue before heading out of town).

I should note that at one point, I was successfully able to see the settings I'd created (in addition to the weather/Mark 1 settings), so I'd had at least minimal success before this error came up.

Beyond the connection issue, I don't think all the types I need are fully supported yet, e.g. the verbose option should be a Boolean, but [the docs](https://mycroft.ai/documentation/skills/skill-settings/) only mentioned support for numbers and strings. Also, the default group value should accept a number, or a string (groups can be identified by their id number, or their name). These are not show stoppers since I can just do string parsing, but figured I'd mention them as well.

I also had trouble getting labels to work for individual fields. Again, not a show stopper, I dumped all the descriptions in a single blob at the top, but figured I'd mention that one as well.

Apologies that I've not opened proper issues for these yet - the plan was to take care of this all when I got home.

Let me know if you have any thoughts, feedback or tips. Happy to do what I can via my phone this week, and pick up the rest when I'm home.

-Chris

**Comment by ChristopherRogers1991** [comment]
Also, I was hoping for basic (verbose, color/brightness step) and advanced (username, ip) sections, but couldn't seem to get multiple sections to work either.

**Comment by penrods** [comment]
Not a rush on this, enjoy your computer-free time!

The checkbox webUI option hasn't been completed yet, but is on the TODO list.  Should happen in the next week or so.

I see a bug with having multiple labels if you don't give the label a 'name' field.  If you want to play with it before we get this fixed, just add "name":"label2", etc.

I also see that there is no UI generated when you define additional groups.  I do see the additional fields, just no indication that they are in a different setting group.  I'll throw a bug on that, too.

**Comment by ChristopherRogers1991** [comment]
Thanks! Any thoughts on the
> Sorry, you don't have skills to configure.
message I get from https://home.mycroft.ai/#/skill?

**Comment by ChristopherRogers1991** [comment]
Ok, I managed to get https://home.mycroft.ai/#/skill working again by deleting and recreating my account (figured if it worked for other people, blowing up my account and starting over would get it working for me again).

I just pushed up a few more changes to the [feature/settingsmeta](https://github.com/ChristopherRogers1991/mycroft-hue/tree/feature/settingsmeta?files=1) branch, and with them, I believe I have this all completed.

I need to do a little more testing, but assuming I don't find any issues, I expect to merge [feature/settingsmeta](https://github.com/ChristopherRogers1991/mycroft-hue/tree/feature/settingsmeta?files=1) tomorrow, or sometime over the weekend. 

@penrods, if you have a few minutes to check it out, please let me know if you have any feedback.

-Chris

**Comment by ChristopherRogers1991** [comment]
Just merged #9, which adds the settingsmeta file, and updates docs/dialogs that referenced old install/configuration methods. Closing this out.
